pkg/upgrade/upgrade.go

```diff
@@ -297,9 +336,63 @@ func CleanupExistingResource(ctx context.Context, cli client.Client, platform de
 		Kind:    "OdhApplication",
 	}
 	multiErr = multierror.Append(multiErr, removOdhApplicationsCR(ctx, cli, JupyterhubApp, "jupyterhub", dscApplicationsNamespace))
+
+	// to take a reference
+	toDelete := jobGetDashboardWatsonResources(dscApplicationsNamespace)
+	multiErr = multierror.Append(multiErr, deleteResources(ctx, cli, &toDelete))
+
 	return multiErr.ErrorOrNil()
 }
 
+func deleteResources(ctx context.Context, c client.Client, jobs *[]JobSpec) error {
+	var errors *multierror.Error
+
+	for _, job := range *jobs {
+		err := deleteOneJob(ctx, c, job)
+		errors = multierror.Append(errors, err)
+	}
+
+	return errors.ErrorOrNil()
+}
+
+func deleteOneJob(ctx context.Context, c client.Client, job JobSpec) error {
```

Comment:
if https://github.com/opendatahub-io/opendatahub-operator/pull/959/files#r1557861353 then this is `deleteResource`

Reply:
> if https://github.com/opendatahub-io/opendatahub-operator/pull/959/files#r1557861353 then this is `deleteResource`

There is name conflict here :)

Reply:
> Updated. Kept `One` in the name
Too fast, variable name probably better to rename as well

Reply:
> Ah right, but then maybe the original one could be renamed to `deleteResourceWithRetry` or sth like that?

Pushed already :) Let's keep it then for further refactoring.